### PR TITLE
Some refactoring

### PR DESCRIPTION
**Pull request by erikzmmrmnn**
- removes DDD folder and instead keeps DDD classes directly in application folder
- moves duplicate code inside DatabaseInterface to private function

### CODE REVIEW

**Comment by erikzmmrmnn** [comment]
I discarded the plan of moving the DDD classes into an own "domain" folder (what I texted you on WhatsApp) because we use use the other classes like Cipher or AbstractDatabaseInterface because we use those classes inside the DDD classes like the EntryRepository for example. Therefore it would contradict the clean architecture structure because the direction of the dependenices would go in the wrong direction.

File: src/application/EntryRepository.h [hunk 1]
```diff
@@ -14,9 +14,9 @@ namespace DDD::Repositories {
         std::set<Entities::Entry> find(ValueObjects::EntryName entryName) const;
         void store(const Entities::Entry &entry) const;
         void remove(const Entities::Entry &entry) const;
-        void modifyEntryName(const Entities::Entry &entry, const ValueObjects::EntryName &newEntryName) const;
-        void modifyLogin(const Entities::Entry &entry, const ValueObjects::Login &newLogin) const;
-        void modifyEncryptedPassword(const Entities::Entry &entry, const ValueObjects::EncryptedPassword &newEncryptedPassword) const;
+        void modifyEntry(const Entities::Entry &entry, const ValueObjects::EntryName &newEntryName) const;
```
Comment (moorts) [inline src/application/EntryRepository.h:17]: Is this based on some design pattern? Seems sort of confusing to me.

File: src/tests/application/encryptor_tests.cpp [hunk 2]
```diff
@@ -1,8 +1,8 @@
-#include <gtest/gtest.h>
-#include "../../../application/DDD/PasswordEncryption.h"
-#include "../../../application/Cipher.h"
-#include "../../../plugins/encryption/CBC_Cipher.h"
-#include "../../../plugins/encryption/BLOCK.h"
+#include "../../cmake-build-debug/_deps/googletest-src/googletest/include/gtest/gtest.h"
```
Comment (moorts) [inline src/tests/application/encryptor_tests.cpp:1]: I'm guessing this fixes the CLion build?

Comment (erikzmmrmnn) [reply src/tests/application/encryptor_tests.cpp:1]: It does not. It was somehow automatically expanded into that from <gtest/gtest.h> by CLion. Thanks for pointing that out, looks pretty ugly.

**Review by moorts — APPROVED** [review]

**Pull request merged**